### PR TITLE
Refactor atom47 and add rev4 and rev5

### PR DESCRIPTION
**Pull request by evyd13**
## Description
Refactors the atom47, adds rev4 and rev5. Rev4 is a very [minimalistic version](https://cdn.discordapp.com/attachments/535222217974480896/707904153577324594/57fj3TV.png) and [rev5 is the opposite](https://cdn.discordapp.com/attachments/535222217974480896/853599021155614770/PXL_20210613_113710540.jpg).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by evyd13** [comment]
This is a breaking change, by the way.

File: keyboards/evyd13/atom47/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
There are merge conflicts here.

**Comment by evyd13** [comment]
Should be solved now!

**Review by drashna — APPROVED** [review]

File: keyboards/evyd13/atom47/readme.md [hunk 2]
```diff
@@ -3,24 +3,13 @@
...
-- 6 Underglow RGB leds  
-- In-switch leds  
-- Through-hole micro usb connector (less likely to break off!)  
-- South facing leds for the QMX/Zealencio users  
-- Multiple layouts  
-- Easily reachable reset button under the spacebar  
-- CapsLock indicator  
+Keyboard Maintainer: [Evelien Dekkers](https://github.com/evyd13)  
+Hardware Supported: Atom47 rev2/rev3/rev4/rev5
+Hardware Availability: [Open source on GitHub](https://github.com/evyd13/atom47)
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/readme.md:8]:
```suggestion
* Keyboard Maintainer: [Evelien Dekkers](https://github.com/evyd13)
* Hardware Supported: Atom47 rev2/rev3/rev4/rev5
* Hardware Availability: [Open source on GitHub](https://github.com/evyd13/atom47)
```

File: keyboards/evyd13/atom47/rev2/rules.mk [hunk 3]
```diff
@@ -1,2 +1,2 @@
-NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+# Build Options
 RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/rev2/rules.mk:2]:
```suggestion
RGBLIGHT_ENABLE = no
```

File: keyboards/evyd13/atom47/rev4/rev4.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  writePinHigh(D1);
+}
+
+bool led_update_kb(led_t led_state) {
+  if(led_update_user(led_state)) {
+    writePin(D1, !led_state.caps_lock);
+  }
+
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/rev4/rev4.c:40]: Shouldn't be needed if you have the define in config.h.

File: keyboards/evyd13/atom47/rev4/rules.mk [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# MCU name
+MCU = atmega32u2
+
+# Build Options
+ENCODER_ENABLE = yes
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/rev4/rules.mk:7]:
```suggestion
BACKLIGHT_ENABLE = no
RGBLIGHT_ENABLE = no
```

File: keyboards/evyd13/atom47/rev5/rev5.c [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    }
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        global_led_state = led_state;
+    }
+    return res;
+}
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/rev5/rev5.c:163]:
```suggestion
void rgb_matrix_indicators_kb(void) {
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(26, 255, 255, 255);
    } else {
        rgb_matrix_set_color(26, 0, 0, 0);
    }
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by tzarc** [comment]
@evyd13 I've pushed some changes that should fix things once merged, can you pull your branch and give them a quick try-out?

**Review by tzarc — APPROVED** [review]

**Comment by evyd13** [comment]
@tzarc I'm getting the following error on building the rev5 with any keymap: https://pastebin.com/gRFDT2eN

**Comment by fauxpark** [comment]
You need to be on the `develop` branch.

**Comment by evyd13** [comment]
> You need to be on the `develop` branch.

Ah, works perfectly now :)

**Pull request merged**